WordPress/src/main/java/org/wordpress/android/ui/reader/repository/usecases/PostLikeUseCase.kt

```diff
@@ -29,86 +24,101 @@ import kotlin.coroutines.Continuation
 import kotlin.coroutines.resume
 
 class PostLikeUseCase @Inject constructor(
-    private val eventBusWrapper: EventBusWrapper,
     private val readerPostActionsWrapper: ReaderPostActionsWrapper,
     private val analyticsUtilsWrapper: AnalyticsUtilsWrapper,
     private val accountStore: AccountStore,
     private val networkUtilsWrapper: NetworkUtilsWrapper,
     @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
 ) {
     private val continuations:
-            MutableMap<PostLikeRequest, Continuation<ReaderRepositoryCommunication>?> = mutableMapOf()
+            MutableMap<PostLikeRequest, Continuation<PostLikeState>?> = mutableMapOf()
 
-    init {
-        eventBusWrapper.register(this)
-    }
+    suspend fun perform(post: ReaderPost, isAskingToLike: Boolean) = flow<PostLikeState> {
+        val wpComUserId = accountStore.account.userId
+        val request = PostLikeRequest(post.postId, post.blogId, isAskingToLike, wpComUserId)
 
-    suspend fun perform(
-        post: ReaderPost,
-        isAskingToLike: Boolean
-    ): ReaderRepositoryCommunication {
-        return withContext(bgDispatcher) {
-            val wpComUserId = accountStore.account.userId
-            val request = PostLikeRequest(post.postId, post.blogId, isAskingToLike, wpComUserId)
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
+            emit(Failed.NoNetwork(request))
```

Comment:
Don't we need to use "return" or "else" branch here? Same goes for `continuations[request] != null`. Wdyt?

Reply:
> Don't we need to use "return" or "else" branch here? Same goes for continuations[request] != null. Wdyt?

Added returns in commit 20a66ac